File: functions/libsodium/variants/c/1_b64_byte_to_urlsafe_char_2.c
#include <stdint.h>

static int b64_byte_to_urlsafe_char(unsigned int x)
{
    char table[] = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_";
    return table[x];
}


int main(){
    b64_byte_to_urlsafe_char(0);
}

File: functions/libsodium/variants/go/1_b64_byte_to_urlsafe_char_2.go
package main

func b64ByteToURLSafeChar(x uint32) int32 {
    if x <= 63 {
        if x < 26 {
            return int32(x + 'A')
        } else if x < 52 {
            return int32(x + 'a' - 26)
        } else if x < 62 {
            return int32(x + '0' - 52)
        } else if x == 62 {
            return '-'
        } else {
            return '_'
        }
    }
    return -1
}


func main(){
    b64ByteToURLSafeChar(0)
}
